WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift

```diff
@@ -70,11 +76,182 @@ struct NewOrder: View {
+    /// Used to change the product quantity in the order.
+    ///
+    struct ProductStepper: View {
+        @State var textSize: CGSize = .zero
+
+        var body: some View {
+            HStack(spacing: textSize.height) {
+                Button {
+                    // TODO: Decrement the product quantity
+                } label: {
+                    Image(uiImage: .minusImage)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: textSize.height)
+                }
+                Text("1") // Fake data - quantity
+                    .background(ViewGeometry())
+                    .onPreferenceChange(ViewSizeKey.self) {
+                        textSize = $0
+                    }
+                Button {
+                    // TODO: Increment the product quantity
+                } label: {
+                    Image(uiImage: .plusImage)
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(height: textSize.height)
+                }
+            }
+            .padding(textSize.height/2)
+            .overlay(
+                RoundedRectangle(cornerRadius: NewOrder.Layout.stepperBorderRadius)
+                    .stroke(Color(UIColor.separator), lineWidth: NewOrder.Layout.stepperBorderWidth)
+            )
+            .accessibilityElement(children: .ignore)
+            .accessibility(label: Text("Quantity"))
+            .accessibility(value: Text("1")) // Fake static data - quantity
+            .accessibilityAdjustableAction { direction in
+                switch direction {
+                case .decrement:
+                    break // TODO: Decrement the product quantity
+                case .increment:
+                    break // TODO: Increment the product quantity
+                @unknown default:
+                    break
+                }
+            }
+        }
+
+        /// Custom preference key to get the size of a view
+        ///
+        struct ViewSizeKey: PreferenceKey {
+            static var defaultValue: CGSize = .zero
+
+            static func reduce(value: inout CGSize, nextValue: () -> CGSize) {
+                value = nextValue()
+            }
+        }
+
+        /// View that calculates its size and sets ViewSizeKey
+        ///
+        /// Used to ensure that stepper button height matches text height, and overlay border is set correctly at larger text sizes.
+        ///
+        struct ViewGeometry: View {
+            var body: some View {
+                GeometryReader { geometry in
+                    Color.clear
+                        .preference(key: ViewSizeKey.self, value: geometry.size)
+                }
+            }
+        }
+    }
+}
+
+// MARK: Custom Views
+/// Represents a button with a plus icon.
+/// Used for any button that adds details to the order.
+///
+private struct AddButton: View {
```

Comment:
As you say, we will be reusing these buttons more and more, I think it makes sense to move this to a `ButtonStyle` so it can be used throughout the app. What do you think?

Reply:
Good call. Added a `PlusButtonStyle` button style in 0252eea.